tip-distributor/src/reclaim_rent_workflow.rs

```diff
@@ -0,0 +1,134 @@
+use {
+    crate::send_transactions_with_retry,
+    anchor_lang::AccountDeserialize,
+    log::info,
+    solana_client::nonblocking::rpc_client::RpcClient,
+    solana_program::pubkey::Pubkey,
+    solana_sdk::{
+        signature::{Keypair, Signer},
+        transaction::Transaction,
+    },
+    std::{error::Error, time::Duration},
+    tip_distribution::{
+        sdk::{
+            derive_config_account_address,
+            instruction::{
+                close_claim_status_ix, close_tip_distribution_account_ix, CloseClaimStatusAccounts,
+                CloseClaimStatusArgs, CloseTipDistributionAccountArgs,
+                CloseTipDistributionAccounts,
+            },
+        },
+        state::{ClaimStatus, Config, TipDistributionAccount},
+    },
+};
+
+pub async fn reclaim_rent(
+    rpc_client: RpcClient,
+    tip_distribution_program_id: Pubkey,
+    signer: Keypair,
+    // Optionally reclaim TipDistributionAccount rents on behalf of validators.
+    should_reclaim_tdas: bool,
+) -> Result<(), Box<dyn Error>> {
+    let accounts = rpc_client
+        .get_program_accounts(&tip_distribution_program_id)
+        .await?;
+
+    let current_epoch = rpc_client.get_epoch_info().await?.epoch;
+
+    let config_pubkey = derive_config_account_address(&tip_distribution_program_id).0;
+    let config_account = rpc_client.get_account(&config_pubkey).await?;
+    let config_account: Config = Config::try_deserialize(&mut config_account.data.as_slice())?;
+
+    let claim_status_accounts: Vec<(Pubkey, ClaimStatus)> = accounts
+        .iter()
+        .filter_map(|(pubkey, account)| {
+            if let Ok(claim_status) = ClaimStatus::try_deserialize(&mut account.data.as_slice()) {
+                Some((*pubkey, claim_status))
+            } else {
+                None
+            }
+        })
+        .filter(|(_, claim_status): &(Pubkey, ClaimStatus)| {
+            // Only return claim statues that we've paid for and ones that are expired to avoid transaction failures.
+            claim_status.claim_status_payer == signer.pubkey()
+                && current_epoch > claim_status.expires_at
```

Comment:
@esemeniuc can't remember if this is inclusive or nah

Reply:
cant remember either :( worst case scenario is we have another epoch of rent sitting around?

Reply:
https://github.com/jito-foundation/jito-programs/blob/master/src/programs/tip-distribution/src/lib.rs#L179